src/middlewares/authenticateRequest.ts

```diff
@@ -0,0 +1,42 @@
+import { Request } from 'express';
+import jwt from 'jsonwebtoken';
+import UserModel from '../models/User';
+
+export default async ({
+  req,
+}: {
+  req: Request;
+}): Promise<AuthContextReturn> => {
+  const authHeader = req.get('Authorization');
+  if (!authHeader) {
+    return { isAuth: false };
```

Comment:
ici, on peut lancer une error

par ex 
` throw new Error("not authenticated");`

Reply:
Oui bien vu ca ferait une meilleure gestion des erreurs 👍
Par contre je pense qu'il faut le faire seulement à partir de la ligne 17 ou la ligne 25 du middleware 
Si on le fait avant, on risque d'avoir une erreur sur chaque requête quand l'utilisateur navigue en mode _Visitor_ 
(Le middleware s'applique à **toutes** les requêtes graphql, et il n'y a pas de token sur toutes les requêtes)